component/metrics/globalrefmap.go

```diff
@@ -89,8 +89,8 @@ func (g *GlobalRefMap) GetOrAddGlobalRefID(l labels.Labels) uint64 {
 	return g.globalRefID
 }
 
-// GetGlobalRefID returns the global refid for a component local combo, or 0 if not found
-func (g *GlobalRefMap) GetGlobalRefID(componentID string, localRefID uint64) uint64 {
+// GetGlobalRefIDForComponent returns the global refid for a component local combo, or 0 if not found
+func (g *GlobalRefMap) GetGlobalRefIDForComponent(componentID string, localRefID uint64) RefID {
```

Comment:
I just noticed that `g.mappings` never gets cleaned up, even after a component goes away, so mappings will leak. Can you add a TODO somewhere to fix the leak?

Reply:
Fixed the leak.